package/MDAnalysis/core/groups.py

```diff
@@ -1039,32 +1039,67 @@ def wrap(self, compound="atoms", center="com", box=None):
             if not all(s == 0.0):
                 o.atoms.translate(s)
 
-    def groupby(self, topattr):
+    def groupby(self, topattrs):
         """Group together items in this group according to values of *topattr*
 
         Parameters
         ----------
-        topattr: str
-           Topology attribute to group components by.
+        topattrs: str or list
+           One or more topology attribute to group components by.
+           Single arguments are passed as a string. Multiple arguments
+           are passed as a list.
 
         Returns
         -------
         dict
-            Unique values of the topology attribute as keys, Groups as values.
+            Unique values of the multiple combinations of topology attributes 
+            as keys, Groups as values.
 
         Example
-        -------
-        To group atoms with the same mass together::
+        -------      
+        To group atoms with the same mass together:     
+                
+        >>> ag.groupby('masses')
+        {12.010999999999999: <AtomGroup with 462 atoms>,
+         14.007: <AtomGroup with 116 atoms>,
+         15.999000000000001: <AtomGroup with 134 atoms>}        
+
+        To group atoms with the same residue name and mass together:
+
+          >>> ag.groupby(['resnames', 'masses'])
+          {('ALA', 1.008): <AtomGroup with 95 atoms>,
+           ('ALA', 12.011): <AtomGroup with 57 atoms>,
+           ('ALA', 14.007): <AtomGroup with 19 atoms>,
+           ('ALA', 15.999): <AtomGroup with 19 atoms>},
+           ('ARG', 1.008): <AtomGroup with 169 atoms>,
+           ...}
+          
+          >>> ag.groupby(['resnames', 'masses'])('ALA', 15.999)
+           <AtomGroup with 19 atoms>
+        
+        .. versionadded:: 0.16.0
+        .. versionchanged:: 0.18.0 The function accepts multiple attributes
+        """
+        
+        res = dict()
+        
+        if isinstance(topattrs, (string_types, bytes)):
+            topattrs = (topattrs,)
+    
+        if isinstance(topattrs[0], bytes):
```

Comment:
So here I learned something new myself. bytes isn't a string type in Python 3, I should have known that. Could be I was a bit to conservative with testing. We can rather just test unicode string and not raw bytes. But let's see how this goes and if it fails for anyone in the future. Should be a rare case.